scheduler/plugin/pool.go

```diff
@@ -59,6 +59,29 @@ func newGuestPool[guest comparable](ctx context.Context, newGuest func(context.C
 	}, nil
 }
 
+// doWithGuest runs initialization functions that precede the scheduling cycle.
+// Notably, only framework.EnqueueExtensions is known to occur at this state.
+func (p *guestPool[guest]) doWithGuest(ctx context.Context, fn func(guest)) (err error) {
```

Comment:
I owe tests on this, but I will wait until feedback and do that with everything else.